script/openqa-clone-custom-git-refspec

```diff
@@ -3,19 +3,32 @@
 
 # Usage:
 #  openqa-clone-custom-git-refspec GITHUB_PR_URL OPENQA_JOB_URL [CUSTOM_TEST_VAR_1=foo] [CUSTOM_TEST_VAR_2=bar]
+#  openqa-clone-custom-git-refspec GITHUB_BRANCH_URL OPENQA_JOB_URL [CUSTOM_TEST_VAR_1=foo] [CUSTOM_TEST_VAR_2=bar]
 #
 # Example:
 #  openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6529 https://openqa.opensuse.org/tests/835060 DESKTOP=textmode
+#  openqa-clone-custom-git-refspec https://github.com/coolgw/os-autoinst-distri-opensuse/tree/nfs https://openqa.opensuse.org/tests/835060 DESKTOP=textmode
 #
 set -o pipefail
 if [ -n "$GITHUB_TOKEN" ]; then
     AUTHENTICATED_REQUEST=" -u $GITHUB_TOKEN:x-oauth-basic "
     echo "Github oauth token provided, peforming authenticated requests"
 fi
 if [ -z ${repo_name+x} ] || [ -z ${pr+x} ]; then
-    pr_url="${1:?"Need 'pr_url' as parameter pointing to a github pull request or 'repo_name' (sending repo) and 'pr' variables, e.g. either 'https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1234' 'me/os-autoinst-distri-opensuse' and '1234'"}"
-    target_repo_part=${pr_url%%/pull*}
-    pr="${pr_url##*/}"
+    first_arg="${1:?"Need 'pr_url' as parameter pointing to a github pull request or 'repo_name' (sending repo) and 'pr' variables, e.g. either 'https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1234' 'me/os-autoinst-distri-opensuse' and '1234' or 'brach'"}"
+    if [[ $first_arg == *"pull"* ]]; then
+        pr_url=$first_arg
+        target_repo_part=${pr_url%%/pull*}
+        pr="${pr_url##*/}"
```

Comment:
the variable `pr` is also used to set the build name (further down)... so if I try this with a github repo instead of a pull request, the `BUILD` variable of the test is set to a very generic name: 

`BUILD=foursixnine/os-autoinst-distri-opensuse#`

use pr or branch name to set a proper build variable

Reply:
1) When we using branch, we have no pr. 
2) For branch name, branch name already exist at the beginning of BUILD string, i do not think we still need this duplicate suffix.   What's your idea? 
BUILD=*foursixnine*/os-autoinst-distri-opensuse#
Reopen it if you still think this is mandatory :)

Reply:
>     1. When we using branch, we have no pr.
> 
>     2. For branch name, branch name already exist at the beginning of BUILD string, i do not think we still need this duplicate suffix.   What's your idea?
>        BUILD=_foursixnine_/os-autoinst-distri-opensuse#
>        Reopen it if you still think this is mandatory :)

If I need to test 4 different pull requests, they would show up in the same build :) possibly, overlapping other results if a branch is reused (like I do with `oopsitsbrokenagain` or if somebody uses `master` all the time).

[BUILD=_foursixnine_/os-autoinst-distri-opensuse#](https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=foursixnine%2Fos-autoinst-distri-opensuse%23)

Reply:
@coolgw ping :)?

Reply:
@foursixnine 
1)If you using branch name as suffix, then we will have BUILD name as:
BUILD=foursixnine/os-autoinst-distri-opensuse#foursixnine
then you push another new change into branch foursixnine and run script again(using branch), the BUILD name still same as before, so add branch name as suffix it not work, correct? 
So, if you need BUILD name will change after every change then we need find correct way to construct the BUILD var.  Correct me if i have any misunderstanding :)
2) I have no idea my PR need still continue or not? Since base my understanding Olive think  we do not need introduce new code but keep using environment variables setting. Thanks for give some advise :)

Reply:
1. when you push further changes into the same branch you can simply retrigger the originally triggered test job
2. You can continue in this PR if you find problems with environment variables or close it again

Reply:
> @foursixnine
> 1)If you using branch name as suffix, then we will have BUILD name as:
> BUILD=foursixnine/os-autoinst-distri-opensuse#foursixnine
> then you push another new change into branch foursixnine and run script again(using branch), the BUILD name still same as before, so add branch name as suffix it not work, correct?
> So, if you need BUILD name will change after every change then we need find correct way to construct the BUILD var. Correct me if i have any misunderstanding :)

Say that the branch name is `oopsitsbrokenagain`, then the `BUILD` variable should be: `BUILD=foursixnine/os-autoinst-distri-opensuse#oopsitsbrokenagain`. Beyond that (i.e separate commits and stuff) it's not really needed :). 

Hope it helps :)

> 2) I have no idea my PR need still continue or not? Since base my understanding Olive think we do not need introduce new code but keep using environment variables setting. Thanks for give some advise :)

I'ts still needed, otherwise you wouldn't have opened it :). Please keep working on it.

Reply:
> > @foursixnine
> > 1)If you using branch name as suffix, then we will have BUILD name as:
> > BUILD=foursixnine/os-autoinst-distri-opensuse#foursixnine
> > then you push another new change into branch foursixnine and run script again(using branch), the BUILD name still same as before, so add branch name as suffix it not work, correct?
> > So, if you need BUILD name will change after every change then we need find correct way to construct the BUILD var. Correct me if i have any misunderstanding :)
> 
> Say that the branch name is `oopsitsbrokenagain`, then the `BUILD` variable should be: `BUILD=foursixnine/os-autoinst-distri-opensuse#oopsitsbrokenagain`. Beyond that (i.e separate commits and stuff) it's not really needed :).
> 
[GW]: Got it :)

> Hope it helps :)
> 
> > 1. I have no idea my PR need still continue or not? Since base my understanding Olive think we do not need introduce new code but keep using environment variables setting. Thanks for give some advise :)
> 
> I'ts still needed, otherwise you wouldn't have opened it :). Please keep working on it.
[GW]:There are two directions keep working on this PR 
1) Add new logic as current implementation(means add suffix for the BUILD name)
2) Using environment variables(not add new logic), but need fix issue
Which one ? @foursixnine

Reply:
@coolgw Let's go on the new logic part :). I think adding environment variable will make things a bit more complicated, unnecessarily

Reply:
you might think so but adding another layer of logic on top of environment variables that are kept and won't work anymore is not necessarily making it easier.

Reply:
> @coolgw Let's go on the new logic part :). I think adding environment variable will make things a bit more complicated, unnecessarily

Updated with new result:
dry_run=echo sh ./openqa-clone-custom-git-refspec os-autoinst/os-autoinst-distri-opensuse#8171 http://openqa.suse.de/tests/3235878
openqa-clone-job --skip-chained-deps --within-instance http://openqa.suse.de 3235878 _GROUP=0 TEST=offline_sles12sp2_ltss_pscc_sdk+we+lp+asmm+contm+lgm+pcm+tcm+wsm+ltss_all_full_x86_64@coolgw/os-autoinst-distri-opensuse#nfs BUILD=coolgw/os-autoinst-distri-opensuse#8171 CASEDIR=https://github.com/coolgw/os-autoinst-distri-opensuse.git#nfs PRODUCTDIR=os-autoinst-distri-opensuse/products/sle NEEDLES_DIR=/var/lib/openqa/cache/openqa.suse.de/tests/sle/products/sle/needles

dry_run=echo sh ./openqa-clone-custom-git-refspec https://github.com/coolgw/os-autoinst-distri-opensuse/tree/nfs http://openqa.suse.de/tests/3235878
openqa-clone-job --skip-chained-deps --within-instance http://openqa.suse.de 3235878 _GROUP=0 TEST=offline_sles12sp2_ltss_pscc_sdk+we+lp+asmm+contm+lgm+pcm+tcm+wsm+ltss_all_full_x86_64@coolgw/os-autoinst-distri-opensuse#nfs BUILD=coolgw/os-autoinst-distri-opensuse#nfs CASEDIR=https://github.com/coolgw/os-autoinst-distri-opensuse.git#nfs PRODUCTDIR=os-autoinst-distri-opensuse/products/sle NEEDLES_DIR=/var/lib/openqa/cache/openqa.suse.de/tests/sle/products/sle/needles
linux-4360:~/openqa/openQA-coolgw/openQA/script #